### ISSUE TITLE
[init-declarations] eslint's JS version of this rule does not handle the case of TS declaration statements. We need a TS version of this rule.

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#1811]
[eslint's init-declarations rule.](https://eslint.org/docs/rules/init-declarations)

**Repro**

```JSON
{
  "rules": {
    "init-declarations": ["error", "always"],
  }
}
```

```TS
declare const foo: number;
```

**Expected Result**

Type declarations should not cause an eslint rule violation.

**Actual Result**

The `init-declarations` rule raises a violation.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.25.0` |
| `@typescript-eslint/parser`        | `2.25.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.4.0` |
| `npm`                              | `6.13.5` |
